### ISSUE TITLE
"No overloads" with `attr.Factory(list)` as default to `attr.ib`.

### ISSUE THREAD

**Issue by jakebailey** [microsoft/pyright#2252]
Pulling this from https://github.com/microsoft/pylance-release/issues/1627.

If I write the code:

```python
import attr

attr.Factory(list)
attr.ib(default=attr.Factory(list))
```

`attr.Factory(list)` alone works, but the second one fails with `No overloads for "Factory" match the provided arguments` at `attr.Factory(list)`.

![image](https://user-images.githubusercontent.com/5341706/131741262-162c5417-dff4-4b90-b590-7e209422eb3f.png)

Perhaps a bi-directional type inference problem?

This reproduces in https://github.com/microsoft/pylance-release/issues/1627's original example, but in that case, it's:

```python
@attr.s(auto_attribs=True)
class C2:
    field1 : int = 0
    field2 : list[Any] = attr.ib(default=attr.Factory(list))
```

Where feasibly, `list[Any]` would be what informs the rest.

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.166, which I just published. It will also be included in the next release of pylance.
